node/node.go

```diff
@@ -24,8 +26,9 @@ type Node struct {
 	// the Node keeps a reference to the DI App that controls the lifecycles of services registered on the Node.
 	app *fx.App
 
-	// the RPC client is only necessary for full nodes
-	RPCClient *rpc.Client `optional:"true"`
+	// CoreClient provides access to Core Node
+	CoreClient    core.Client `optional:"true"`
+	CoreRPCClient *rpc.Client `optional:"true"`
```

Comment:
What's the point of having both then?

Reply:
No point :) But we decided to postpone this conversation to BlockService PR.

Honestly, now I see that I should've started describing [this](https://github.com/celestiaorg/celestia-node/pull/48#pullrequestreview-747832676) only after fledging our this PR, as now my message should be much clearer.

Reply:
Both in general or both in the Node?

Reply:
both in general and inside the node -- this PR makes it so that we can remove the `rpc.Client`, so lets just remove it and use `Client` interface from the embedded core proc. Can you pls do this in this PR, let's merge it asap and then I'll rebase block service on top. Does that sound good @Wondertan ?

Reply:
Done